examples/visual/car_sequencing.ipynb

```diff
@@ -0,0 +1,302 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "ongoing-target",
+   "metadata": {},
+   "source": [
+    "# Car Sequencing problem in CPMPy, with visualization\n",
+    "\n",
+    "Based on Alexander Schiendorfer's, but with simplified visualization\n",
+    "https://github.com/Alexander-Schiendorfer/cp-examples/tree/main/car-sequencing\n",
+    "\n",
+    "Given different types of cars (based on the present options), an overall car demand per type \n",
+    "and some constraints on how many times properties can be scheduled in consecutive timeslots,\n",
+    "the program finds a feasible sequencing of the different cars for a timetable.\n",
+    "\n",
+    "This problem is also described in CSPlib: https://www.csplib.org/Problems/prob001/"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "universal-extent",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Academic license - for non-commercial use only - expires 2022-10-21\n",
+      "Using license file /home/tias/local/src/gurobi912/linux64/../gurobi.lic\n"
+     ]
+    }
+   ],
+   "source": [
+    "import pandas as pd\n",
+    "import matplotlib.pyplot as plt\n",
+    "import matplotlib.colors as mcolors\n",
+    "from cpmpy import *"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "worldwide-spell",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# the CPMpy model and variables\n",
+    "def model_car_sequence(demand, per_slots, at_most, requires):\n",
+    "    nSlots = sum(demand)  # The amount of timeslots to be filled\n",
+    "    nOptions = len(at_most)  # The amount of different options\n",
+    "    nCarConfigs = len(demand)  # The amount of different car types\n",
+    "    requires = cpm_array(requires)  # For element constraint\n",
+    "\n",
+    "    # Decision variables\n",
+    "    slots = intvar(1, nCarConfigs, shape=nSlots, name=\"slots\")  # Sequence of different car types\n",
+    "    setup = boolvar(shape=(nSlots, nOptions), name=\"setup\")  # Sequence of different options based on the car type\n",
+    "\n",
+    "    m = Model()\n",
+    "\n",
+    "    # The amount of each type of car in the sequence has to be equal to the demand for that type\n",
+    "    # (offset because we talk about car type 1->nCarConfigs)\n",
+    "    m += [sum(slots == (i+1)) == demand[i] for i in range(nCarConfigs)]\n",
+    "\n",
+    "    # Make sure that the options in the setup table correspond to those of the car type\n",
+    "    for s in range(nSlots):\n",
+    "        m += [setup[s,o] == requires[slots[s]-1,o] for o in range(nOptions)]\n",
+    "\n",
+    "    # Check that no more than \"at most\" car properties are used per \"per_slots\" slots\n",
+    "    for o in range(nOptions):\n",
+    "        for s in range(nSlots - per_slots[o]):\n",
+    "            slotrange = range(s, s + per_slots[o])\n",
```

Comment:
In the csp-lib version I used the `rolling_window` function from numpy. For me it's more clear using that one but this is fine too